app/controllers/TaxsLanguageController.scala

```diff
@@ -17,17 +17,18 @@
 package controllers
 
 import com.google.inject.Inject
-import play.api.i18n.{I18nSupport, Lang, MessagesApi}
-import play.api.mvc.{Action, AnyContent}
+import play.api.i18n.Lang
+import play.api.mvc.{Action, AnyContent, MessagesControllerComponents}
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import uk.gov.hmrc.play.partials.FormPartialRetriever
 
-class TaxsLanguageController @Inject()(val messagesApi: MessagesApi)(implicit val formPartialRetriever: FormPartialRetriever) extends FrontendController with I18nSupport {
+class TaxsLanguageController @Inject()(mcc : MessagesControllerComponents)(implicit val formPartialRetriever: FormPartialRetriever) extends FrontendController(mcc) {
 
   def switchLanguage(lang: String): Action[AnyContent] = Action { implicit request =>
     request.headers.get(REFERER) match {
-      case Some(referrer) => Redirect(referrer).withLang(Lang(lang))
-      case _              => Redirect(routes.IndexController.authorisedIndex()).withLang(Lang(lang))
+        //TODO look for the logic to swtich the language
```

Comment:
Did you find it? Is this still relevant?

Reply:
yep handled the language switch logic as part of code clean up